controller/volume_controller.go

```diff
@@ -2916,7 +2916,21 @@ func (vc *VolumeController) checkForAutoDetachment(v *longhorn.Volume, e *longho
 			break
 		}
 	}
-	if !(e.Spec.RequestedBackupRestore != "" && e.Spec.RequestedBackupRestore == e.Status.LastRestoredBackup &&
+
+	// make sure the backup volume has been synced at least once
+	if backupVolumeName, isExist := v.Labels[types.LonghornLabelBackupVolume]; isExist && backupVolumeName != "" {
+		backupVolume, err := vc.ds.GetBackupVolumeRO(backupVolumeName)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get backup volume: %v", v.Name)
+		}
+		if backupVolume.Status.LastSyncedAt.Before(backupVolume.Spec.SyncRequestedAt.DeepCopy()) {
```

Comment:
Is the `DeepCopy()` necessary?

Reply:
Why do we need to check it be synced at least once? Can you add a comment here?

Reply:
yeah, `DeepCopy` is unnecessary.

Reply:
Thanks, you are right, there is no need to pass the DeepCopy object. 

Since the parameter type of `v1.time.Before` is `*Time`, I passed the address of `SyncRequestedAt`.